server/src/operators/dataset_operator.rs

```diff
@@ -340,14 +342,27 @@ pub async fn get_datasets_by_organization_id(
         .await
         .map_err(|_| ServiceError::BadRequest("Could not get database connection".to_string()))?;
 
-    let dataset_and_usages: Vec<(Dataset, DatasetUsageCount)> = datasets_columns::datasets
-        .inner_join(dataset_usage_counts_columns::dataset_usage_counts)
-        .filter(datasets_columns::deleted.eq(0))
-        .filter(datasets_columns::organization_id.eq(org_id.into_inner()))
-        .select((Dataset::as_select(), DatasetUsageCount::as_select()))
-        .load::<(Dataset, DatasetUsageCount)>(&mut conn)
-        .await
-        .map_err(|_| ServiceError::NotFound("Could not find organization".to_string()))?;
+    let dataset_and_usages = match pagination.limit {
+        Some(limit) => datasets_columns::datasets
+            .inner_join(dataset_usage_counts_columns::dataset_usage_counts)
+            .filter(datasets_columns::deleted.eq(0))
+            .filter(datasets_columns::organization_id.eq(org_id.into_inner()))
+            .limit(limit.into())
+            .offset(pagination.offset.unwrap_or(0))
+            .order(datasets_columns::created_at.desc())
+            .select((Dataset::as_select(), DatasetUsageCount::as_select()))
+            .load::<(Dataset, DatasetUsageCount)>(&mut conn)
+            .await
+            .map_err(|_| ServiceError::NotFound("Could not find organization".to_string()))?,
+        None => datasets_columns::datasets
+            .inner_join(dataset_usage_counts_columns::dataset_usage_counts)
+            .filter(datasets_columns::deleted.eq(0))
+            .order(datasets_columns::created_at.desc())
+            .select((Dataset::as_select(), DatasetUsageCount::as_select()))
+            .load::<(Dataset, DatasetUsageCount)>(&mut conn)
+            .await
+            .map_err(|_| ServiceError::NotFound("Could not find organization".to_string()))?,
+    };
```

Comment:
is there a way to build the queries more dynamically here?

Reply:
Yes, `into_boxed` would let you achieve that. 

Check this out - https://github.com/devflowinc/trieve/blob/main/server/src/operators/dataset_operator.rs#L104

Reply:
Did it in my commit